tools/integration_tests/stale_handle/stale_file_handle_synced_file_test.go

```diff
@@ -0,0 +1,214 @@
+
+const Content = "foobar"
+const Content2 = "foobar2"
+
+type staleFileHandleSyncedFile struct {
+	flags []string
+	suite.Suite
+}
+
+func (s *staleFileHandleSyncedFile) SetupSuite() {
+	mountGCSFuseAndSetupTestDir(s.flags, ctx, storageClient, testDirName)
+}
+
+func (s *staleFileHandleSyncedFile) TearDownSuite() {
+	setup.UnmountGCSFuse(rootDir)
+}
+
+////////////////////////////////////////////////////////////////////////
+// Tests
+////////////////////////////////////////////////////////////////////////
+
+func (s *staleFileHandleSyncedFile) TestSyncedObjectClobberedRemotelyReadThrowsStaleFileHandleError() {
+	testCaseDir := "TestSyncedObjectClobberedRemotelyReadThrowsStaleFileHandleError"
+	targetDir := path.Join(testDirPath, testCaseDir)
+	operations.CreateDirectory(targetDir, s.T())
+	// Create an object on bucket
+	err := CreateObjectOnGCS(ctx, storageClient, path.Join(testDirName, testCaseDir, FileName1), GCSFileContent)
+	assert.NoError(s.T(), err)
+	filePath := path.Join(targetDir, FileName1)
+	// Open the read handle
+	fh, err := operations.OpenFileAsReadonly(filePath)
+	assert.NoError(s.T(), err)
+	// Replace the underlying object with a new generation.
+	err = WriteToObject(ctx, storageClient, path.Join(testDirName, testCaseDir, FileName1), FileContents, storage.Conditions{})
+	assert.NoError(s.T(), err)
+
+	buffer := make([]byte, GCSFileSize)
+	err = operations.ReadBytesFromFile(fh, GCSFileSize, buffer)
+
+	operations.ValidateStaleNFSFileHandleError(s.T(), err)
+	ValidateObjectContentsFromGCS(ctx, storageClient, path.Join(testDirName, testCaseDir), FileName1, FileContents, s.T())
+}
+
+func (s *staleFileHandleSyncedFile) TestSyncedObjectClobberedRemotelyFirstWriteThrowsStaleFileHandleError() {
+	testCaseDir := "TestSyncedObjectClobberedRemotelyFirstWriteThrowsStaleFileHandleError"
+	targetDir := path.Join(testDirPath, testCaseDir)
+	operations.CreateDirectory(targetDir, s.T())
+	err := CreateObjectOnGCS(ctx, storageClient, path.Join(testDirName, testCaseDir, FileName1), GCSFileContent)
+	assert.NoError(s.T(), err)
+	filePath := path.Join(targetDir, FileName1)
+	fh, err := operations.OpenFileAsWriteOnly(filePath)
+	assert.NoError(s.T(), err)
+	// Replace the underlying object with a new generation.
+	err = WriteToObject(ctx, storageClient, path.Join(testDirName, testCaseDir, FileName1), FileContents, storage.Conditions{})
+	assert.NoError(s.T(), err)
+
+	_, err = fh.WriteString(Content)
+
+	operations.ValidateStaleNFSFileHandleError(s.T(), err)
+	// Attempt to sync to file should not result in error as we first check if the
+	// content has been dirtied before clobbered check in Sync flow.
+	operations.SyncFile(fh, s.T())
+	// Validate that object is not updated with new content as write failed.
+	ValidateObjectContentsFromGCS(ctx, storageClient, path.Join(testDirName, testCaseDir), FileName1, FileContents, s.T())
+}
+
+func (s *staleFileHandleSyncedFile) TestSyncedObjectClobberedRemotelySyncAndCloseThrowsStaleFileHandleError() {
```

Comment:
This test is common to local file as well. can be extracted to a common test suite.